assets/blocks/contact-teacher-block/index.js

```diff
@@ -20,15 +25,19 @@ export default createButtonBlockType( {
 			'sensei-lms'
 		),
 		title: __( 'Contact Teacher', 'sensei-lms' ),
-		attributes: {
-			text: {
-				default: __( 'Contact Teacher', 'sensei-lms' ),
-			},
-		},
+		attributes,
 		styles: [
 			BlockStyles.Fill,
 			{ ...BlockStyles.Outline, isDefault: true },
 			BlockStyles.Link,
 		],
+		deprecated: [
+			{
+				attributes,
+				save() {
+					return <></>;
```

Comment:
I liked it to avoid code duplication, but I wonder if a deprecation of something that's not exactly deprecated is a good practice. Also, going ahead with that, `null` wouldn't make more sense instead of an empty fragment?

Reply:
It's primarily there to support the empty form of the block, so we don't have to insert the generated content into the template, which would also make the text label hard to translate. The 'deprecated' setting is a bit awkwardly named, but I think it's an okay fit. I'm not sure it would work with null, the goal is to support the 'empty' generated content case, so this seemed appropriate.